examples/Cassie/osc_walk/high_level_command.cc

```diff
@@ -0,0 +1,207 @@
+                    .get_index();
+  yaw_port_ = this->DeclareVectorOutputPort(BasicVector<double>(1),
+                                            &HighLevelCommand::CopyHeadingAngle)
+                  .get_index();
+  xy_port_ =
+      this->DeclareVectorOutputPort(BasicVector<double>(2),
+                                    &HighLevelCommand::CopyDesiredHorizontalVel)
+          .get_index();
+
+  // Declare update event
+  DeclarePerStepDiscreteUpdateEvent(&HighLevelCommand::DiscreteVariableUpdate);
+
+  // Discrete state which stores previous timestamp
+  prev_time_idx_ = DeclareDiscreteState(VectorXd::Zero(1));
+
+  // Discrete state which stores the desired yaw velocity
+  des_yaw_vel_idx_ = DeclareDiscreteState(VectorXd::Zero(1));
+
+  // Discrete state which stores the desired horizontal velocity
+  des_horizontal_vel_idx_ = DeclareDiscreteState(VectorXd::Zero(2));
+}
+
+EventStatus HighLevelCommand::DiscreteVariableUpdate(
+    const Context<double>& context,
+    DiscreteValues<double>* discrete_state) const {
+  double current_time = context.get_time();
+  auto prev_time =
+      discrete_state->get_mutable_vector(prev_time_idx_).get_mutable_value();
+
+  if (current_time > prev_time(0)) {
+    prev_time(0) = current_time;
+
+    // Read in current state
+    const OutputVector<double>* robotOutput =
+        (OutputVector<double>*)this->EvalVectorInput(context, state_port_);
+    VectorXd q = robotOutput->GetPositions();
+    VectorXd v = robotOutput->GetVelocities();
+
+    // Kinematics cache and indices
+    KinematicsCache<double> cache = tree_.CreateKinematicsCache();
+    // Modify the quaternion in the begining when the state is not received from
+    // the robot yet
+    // Always remember to check 0-norm quaternion when using doKinematics
+    multibody::SetZeroQuaternionToIdentity(&q);
+    cache.initialize(q);
+    tree_.doKinematics(cache);
+
+    // Get center of mass position and velocity
+    Vector3d com_pos = tree_.centerOfMass(cache);
+    MatrixXd J = tree_.centerOfMassJacobian(cache);
+    Vector3d com_vel = J * v;
+
+    //////////// Get desired yaw velocity ////////////
+    // Get approximated heading angle of pelvis
+    Vector3d pelvis_heading_vec =
+        tree_.CalcBodyPoseInWorldFrame(cache, tree_.get_body(pelvis_idx_))
+            .linear()
+            .col(0);
+    double approx_pelvis_yaw =
+        atan2(pelvis_heading_vec(1), pelvis_heading_vec(0));
+
+    // Get desired heading angle of pelvis
+    Vector2d global_com_pos_to_target_pos =
+        global_target_position_ - com_pos.segment(0, 2);
+    double desired_yaw =
+        atan2(global_com_pos_to_target_pos(1), global_com_pos_to_target_pos(0));
+
+    // Get current yaw velocity
+    double yaw_vel = v(2);
+
+    // PD position control
+    double des_yaw_vel =
+        kp_yaw_ * (desired_yaw - approx_pelvis_yaw) + kd_yaw_ * (-yaw_vel);
+    des_yaw_vel = std::min(vel_max_yaw_, std::max(vel_min_yaw_, des_yaw_vel));
```

Comment:
Minor comment - when we switch over to c++17, we should use [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) instead of these min(max) statements

Reply:
Sounds good!